lib/ansible/playbook/play_context.py

```diff
@@ -415,7 +416,8 @@ def detect_su_prompt(data):
             else:
                 raise AnsibleError("Privilege escalation method not found: %s" % self.become_method)
 
-            self.prompt      = prompt
+            if self.become_pass:
+                self.prompt = prompt
             self.success_key = success_key
```

Comment:
You need to always set self.prompt here.  Since the code sets prompt  to None at the top of the file, I think self.prompt = prompt would work.  Otherwise, just an ```else: self.prompt = None``` would work.